pyodide-build/pyodide_build/pywasmcross.py

```diff
@@ -358,14 +357,55 @@ def replay_genargs_handle_argument(arg: str) -> str | None:
     return arg
 
 
-def calculate_exports(line: list[str], export_all: bool) -> Iterator[str]:
-    """
-    Collect up all the object files and archive files being linked and list out
-    symbols in them that are marked as public. If ``export_all`` is ``True``,
-    then return all public symbols. If not, return only the public symbols that
-    begin with `PyInit`.
-    """
-    objects = [arg for arg in line if arg.endswith(".a") or arg.endswith(".o")]
+def calculate_object_exports_readobj(objects: list[str]) -> list[str] | None:
+    which_emcc = shutil.which("emcc")
+    assert which_emcc
+    emcc = Path(which_emcc)
+    args = [
+        str((emcc / "../../bin/llvm-readobj").resolve()),
+        "--section-details",
+        "-st",
+    ] + objects
+    completedprocess = subprocess.run(
+        args, encoding="utf8", capture_output=True, env={"PATH": os.environ["PATH"]}
+    )
+    if completedprocess.returncode:
+        print(f"Command '{' '.join(args)}' failed. Output to stderr was:")
+        print(completedprocess.stderr)
+        sys.exit(completedprocess.returncode)
+
+    if "bitcode files are not supported" in completedprocess.stderr:
+        return None
+
+    result = []
+    insymbol = False
+    for line in completedprocess.stdout.split("\n"):
```

Comment:
Mybe better to move this to a private helper function (that takes in str and returns the result) and additionally give it a few stdout examples in doctests. Would be easier to understand what kind of stdout output we are typically getting and what is done with it.

Reply:
This command has a json output format... but frustratingly it outputs not quite valid json. The errors are minor enough that it seemed like it might be possible to fix by find and replace after the fact but it wasn't worth it.